src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -20,12 +21,14 @@ public sealed class JobHostConfiguration : IServiceProvider
     {
         private static readonly IConsoleProvider ConsoleProvider = new DefaultConsoleProvider();
 
+        // In some legacy cases, JobHostConfiguration specifically exposes properties on the derived object 
+        // (and not on the core interface in the _services collection).  
```

Comment:
Sure. I'll remove it.  But the point I'm making is that there's a gross inconsistency in here that JobHsotConfig is an IServiceProvider, but has several objects that are outside the service provider for now good reason. For example, `DefaultStorageAccountProvider _storageAccountProvider`.  DefaultStorageAccountProvider implements IStorageAccountProvider, but JobHostConfig exposes the derived class and accesses properties on DefaultStorageAccountProvider that aren't on IStorageAccountProvider.